.github/workflows/nightly-build-and-test.yml

```diff
@@ -0,0 +1,38 @@
+name: Nightly Build and Test
+
+on:
+  schedule:
+    - cron: "0 0 * * *" # Runs at midnight every day
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Get latest vllm tag
+        id: latest-vllm-tag
+        run: |
+          git fetch https://github.com/vllm-project/vllm.git --tags 
+          LATEST_TAG=$(git describe --tags $(git rev-list --tags --max-count=1))
+          echo "LATEST_TAG=${LATEST_TAG:1}" >> $GITHUB_OUTPUT
+      - name: Update vllm version to latest
+        env:
+          VLLM_TAG: ${{ steps.latest-vllm-tag.outputs.LATEST_TAG}}
+        run: |
+          echo $VLLM_TAG
+          python -m pip install --upgrade pip
+          pip install tomli-w
+          python scripts/update_vllm_version.py --vllm_version $VLLM_TAG
```

Comment:
should we short circuit with a success and exit if we are already on latest version ?